src/connection.cpp

```diff
@@ -154,6 +154,8 @@ static bool ApplyPreconnAttrs(HDBC hdbc, SQLINTEGER ikey, PyObject *value, char
         return false;
     }
 
+    TRACE("ApplyPreconnAttrs.SQLSetConnectAttrW key=%d value=%p vallen=%d\n", ikey, ivalue);
```

Comment:
I haven't tested this locally, but it appears there are 3 placeholders and 2 values.